source/guides/references/protractor-migration-guide.md

```diff
@@ -0,0 +1,568 @@
+
+**Before: Protractor**
+
+```jsx
+// Click on the element
+element(by.css('button')).click()
+
+// Send keys to the element (usually an input)
+element(by.css('input')).sendKeys('my text')
+
+// Clear the text in an element (usually an input).
+element(by.css('input')).clear()
+
+// Get the attribute of an element
+// Example: Get the value of an input
+element(by.css('input')).getAttribute('value')
+```
+
+**After: Cypress**
+
+```jsx
+// Click on the element
+cy.get('button').click()
+
+// Send keys to the element (usually an input)
+cy.get('input').type('my text')
+
+// Clear the text in an element (usually an input)
+cy.get('input').clear()
+
+// Get the attribute of an element
+// Example: Get the value of an input
+cy.get('input').its('value')
+```
+
+You can learn more about [interacting with DOM elements in our official documentation](https://docs.cypress.io/guides/core-concepts/interacting-with-elements.html).
+
+### Assertions
+
+Similar to Protractor, Cypress enables to use human readable assertions.
+
+**Before: Protractor**
+
+```js
+describe('verify elements on a page', () => {
+  it('verifies that a link is visible', () => {
+    expect($('a.submit-link').isDisplayed()).toBe(true)
+  })
+})
+```
+
+**After: Cypress**
+
+```js
+describe('verify elements on a page', () => {
+	it('verifies that a link is visible', () => {
+		cy.get('a.submit-link').should('be.visible)
+	})
+})
+```
+
+But Cypress has one additional feature that can make a critical difference in the reliability of your tests' assertions: [retry-ability](https://docs.cypress.io/guides/core-concepts/retry-ability.html). When your test fails an assertion or command, Cypress will mimic a real user with build-in wait times and multiple attempts at asserting your tests in order to minimize the amount of false negatives / positives. 
+
+In the example above, if the submit link does not appear on the page at the exact moment when Protractor runs the test (which can be due to any number of factors including API calls, slow browser rendering, etc.), your test will fail. However, Cypress factors these conditions into its assertions and will only fail if the time goes beyond a reasonable amount.
+
+You can learn more about how Cypress handles [assertions in our official documentation](https://docs.cypress.io/guides/references/assertions.html).
+
+### Network Spies
+
+By default, Protractor utilizes Jasmine's `spyOn` function to call a method and test that it was called.
+
+```js
+// paymentApi.js
+export default {
+  submit() { ... }
+}
+```
+
+**Before: Protractor**
+
+```js
+// Protractor
+import paymentApi from './paymentApi.js'
+
+describe('Protractor payment example', () => {
+  it('verifies payment is only sent once', () => {
+    spyOn(paymentApi, 'submit')
+    browser.get('/payment')
+    expect(paymentApi.submit).toHaveBeenCalledTimes(1)
+  })
+})
+```
+
+Fortunately for us, creating a spy in Cypress is almost identical!
+
+**After: Cypress**
+
+```js
+// Cypress
```

Comment:
Is this redundant with header on line 362?

Reply:
Good catch!